expression/column.go

```diff
@@ -40,6 +41,22 @@ func (col *CorrelatedColumn) Clone() Expression {
 	return col
 }
 
+// HashCode implements Expression interface.
+func (col *CorrelatedColumn) HashCode(sc *stmtctx.StatementContext) []byte {
```

Comment:
IMHO, there is not need to implement this method for `CorrelatedColumn`. It can be regared as `Column`.